framework/src/modules/chain/components/storage/entities/chain_meta.js

```diff
@@ -0,0 +1,183 @@
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const path = require('path');
+const assert = require('assert');
+const { defaults, pick } = require('lodash');
+const {
+	entities: { BaseEntity },
+	utils: {
+		filterTypes: { BOOLEAN },
+	},
+} = require('../../../../../components/storage');
+
+const sqlFiles = {
+	upsert: 'chain_meta/upsert.sql',
+	get: 'chain_meta/get.sql',
+	delete: 'chain_meta/delete.sql',
+};
+
+/**
+ * ChainMeta
+ * @typedef {Object} ChainMeta
+ * @property {string} key
+ * @property {string} value
+ */
+
+/**
+ * ChainMeta Filters
+ * @typedef {Object} filters.ChainMeta
+ * @property {string} [key]
+ * @property {string} [key_eql]
+ * @property {string} [key_ne]
+ */
+
+class ChainMeta extends BaseEntity {
+	constructor(adapter, defaultFilters = {}) {
+		super(adapter, defaultFilters);
+
+		this.addField('key', 'string', { filter: BOOLEAN });
+		this.addField('value', 'string');
+
+		this.sqlDirectory = path.join(path.dirname(__filename), '../sql');
+		this.SQLs = this.loadSQLFiles('chain_meta', sqlFiles, this.sqlDirectory);
+	}
+
+	/**
+	 * Get list of meta information
+	 *
+	 * @param {filters.ChainMeta|filters.ChainMeta[]} [filters = {}]
+	 * @param {Object} [options = {}] - Options to filter data
+	 * @param {Number} [options.limit=10] - Number of records to fetch
+	 * @param {Number} [options.offset=0] - Offset to start the records
+	 * @param {Object} [tx] - Database transaction object
+	 * @return {Promise.<ChainMeta[], Error>}
+	 */
+	get(filters = {}, options = {}, tx = null) {
+		return this._getResults(filters, options, tx);
+	}
+
+	/**
+	 * Get list of meta information
+	 *
+	 * @param {filters.ChainMeta|filters.ChainMeta[]} [filters = {}]
+	 * @param {Object} [options = {}] - Options to filter data
+	 * @param {Number} [options.limit=10] - Number of records to fetch
+	 * @param {Number} [options.offset=0] - Offset to start the records
+	 * @param {Object} [tx] - Database transaction object
+	 * @return {Promise.<ChainMeta, Error>}
+	 */
+	getOne(filters = {}, options = {}, tx = null) {
+		const expectedResultCount = 1;
+		return this._getResults(filters, options, tx, expectedResultCount);
+	}
+
+	/**
+	 * Fetch the key value or resolve to null
+	 *
+	 * @param {string} key - Key to fetch
+	 * @param {Object} [tx] - Database transaction object
+	 * @return {Promise<{key: string, value: string} | {}>}
+	 */
+	async getKey(key, tx) {
+		assert(key, 'Must provide the key to get');
+
+		return this.getOne({ key }, {}, tx)
+			.then(data => data.value)
+			.catch(error => {
+				if (error.message === 'No data returned from the query.') {
```

Comment:
No better way to check this? ie) check with error type etc
and do we really want to catch this error?

Reply:
Yes we want to return null in that case so this check is needed. Othercase is to get list of records and based on length decide the return value.

Reply:
but getOne is throwing error, but getKey is not throwing when no key found?
Both are pretty much same feature, but seems inconsistent

Reply:
We need different behaviour for convenient use so have different interfaces.